src/compiler/emitter.ts

```diff
@@ -596,7 +596,21 @@ export function getOutputDeclarationFileName(inputFileName: string, configFile:
     );
 }
 
-function getOutputJSFileName(inputFileName: string, configFile: ParsedCommandLine, ignoreCase: boolean, getCommonSourceDirectory?: () => string) {
+/** @internal */
+export function getOutputDeclarationFileNameWithoutConfigFile(inputFileName: string, options: CompilerOptions, ignoreCase: boolean, currentDirectory: string, getCommonSourceDirectory: () => string) {

```

Comment:
Why do we need separate functions for these. whats different. i couldnt make out from the function,
configFile used in the helper method only needs options and filenames (root file names) so we have that in program so we could change to `Pick<ParsedCommandLine, "options" | "fileNames">` and that should work right. or am i missing something.